cmd/erasure-server-pool.go

```diff
@@ -1773,3 +1773,33 @@ func (z *erasureServerPools) GetObjectTags(ctx context.Context, bucket, object s
 
 	return z.serverPools[idx].GetObjectTags(ctx, bucket, object, opts)
 }
+
+// TransitionObject - transition object content to target tier.
+func (z *erasureServerPools) TransitionObject(ctx context.Context, bucket, object string, opts ObjectOptions) error {
+	object = encodeDirObject(object)
+	if z.SinglePool() {
+		return z.serverPools[0].TransitionObject(ctx, bucket, object, opts)
+	}
+	// We don't know the size here set 1GiB atleast.
+	idx, err := z.getPoolIdx(ctx, bucket, object, 1<<30)
```

Comment:
Aren't we transitioning FROM this `z`? If so, we don't need free space, and we just need the index of the existing object.
```suggestion
	idx, err := z.getPoolIdx(ctx, bucket, object, 0)
```
Added https://github.com/minio/minio/pull/11799 which is even more appropriate.

Reply:
Thanks @klauspost for the PR. I have rebased to use getPoolIdxExisting.